integration/service/oasis/stationgraph/node_container.go

```diff
@@ -0,0 +1,76 @@
+package stationgraph
+
+import "github.com/Telenav/osrm-backend/integration/service/oasis/chargingstrategy"
+
+type logicNodeIdentifier2NodePtr map[logicNodeIdentifier]*node
+type nodeID2NodePtr map[nodeID]*node
+type nodeID2StationID map[nodeID]string
+
+type nodeContainer struct {
+	logicNode2NodePtr logicNodeIdentifier2NodePtr
+	id2NodePtr        nodeID2NodePtr
+	id2StationID      nodeID2StationID
+	counter           int
+}
+
+func newNodeContainer() *nodeContainer {
+	return &nodeContainer{
+		logicNode2NodePtr: make(logicNodeIdentifier2NodePtr),
+		id2NodePtr:        make(nodeID2NodePtr),
+		id2StationID:      make(nodeID2StationID),
+		counter:           0,
+	}
+}
+
+func (nc *nodeContainer) addNode(stationID string, targetState chargingstrategy.State, location locationInfo) *node {
+	key := logicNodeIdentifier{stationID, targetState}
+
+	if n, ok := nc.logicNode2NodePtr[key]; ok {
+		return n
+	} else {
+		n = &node{
+			id: (nodeID(nc.counter)),
+			chargeInfo: chargeInfo{
+				arrivalEnergy: 0.0,
+				chargeTime:    0.0,
+				targetState:   targetState,
+			},
+			locationInfo: location,
+		}
+		nc.logicNode2NodePtr[key] = n
+		nc.id2NodePtr[n.id] = n
+		nc.id2StationID[n.id] = stationID
+		nc.counter++
+
+		return n
+	}
+}
+
+func (nc *nodeContainer) getNode(id nodeID) *node {
+	if n, ok := nc.id2NodePtr[id]; ok {
+		return n
+	} else {
```

Comment:
Prefer to remove the `else`. See https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow

Reply:
agree.